server/dirstore.go

```diff
@@ -0,0 +1,706 @@
+		}
+		if err = os.MkdirAll(dirPath, 0755); err != nil {
+			return "", err
+		}
+		if fullPath, err = validateDirPath(dirPath); err != nil {
+			return "", err
+		}
+	}
+	return fullPath, nil
+}
+
+func NewExpiringDirJWTStore(dirPath string, shard bool, create bool, expireCheck time.Duration, limit int64,
+	evictOnLimit bool, ttl time.Duration, changeNotification JWTChanged) (*DirJWTStore, error) {
+	fullPath, err := newDir(dirPath, create)
+	if err != nil {
+		return nil, err
+	}
+	theStore := &DirJWTStore{
+		directory: fullPath,
+		shard:     shard,
+		changed:   changeNotification,
+	}
+	if expireCheck <= 0 {
+		if ttl != 0 {
+			expireCheck = ttl / 2
+		}
+		if expireCheck == 0 || expireCheck > time.Minute {
+			expireCheck = time.Minute
+		}
+	}
+	if limit <= 0 {
+		limit = math.MaxInt64
+	}
+	theStore.startExpiring(expireCheck, limit, evictOnLimit, ttl)
+	theStore.Lock()
+	err = filepath.Walk(dirPath, func(path string, info os.FileInfo, err error) error {
+		if strings.HasSuffix(path, extension) {
+			if theJwt, err := ioutil.ReadFile(path); err == nil {
+				hash := sha256.Sum256(theJwt)
+				_, file := filepath.Split(path)
+				theStore.expiration.Track(strings.TrimSuffix(file, "."+extension), &hash, string(theJwt))
+			}
+		}
+		return nil
+	})
+	theStore.Unlock()
+	if err != nil {
+		theStore.Close()
+		return nil, err
+	}
+	theStore.readonly = false
+	return theStore, err
+}
+
+// NewDirJWTStore returns an empty, mutable directory-based JWT store
+func NewDirJWTStore(dirPath string, shard bool, create bool, changeNotification JWTChanged, errorNotification JWTError) (*DirJWTStore, error) {
+	// Set evictOnLimit to false and set a limit that can't be meet.
+	// This will cause LRU to not be updated, so it can be used as a list to iterate and find deleted files
+	// Set expiration check to never happen
+	st, err := NewExpiringDirJWTStore(dirPath, shard, create, time.Duration(math.MaxInt64), math.MaxInt64, false, time.Duration(math.MaxInt64), nil)
+	if err != nil {
+		return nil, err
+	}
+	st.Lock()
+	st.changed = changeNotification
+	st.errorOccurred = errorNotification
+	st.Unlock()
+	if err := st.startWatching(); err != nil {
+		st.Close()
+		return nil, err
+	} else {
+		return st, nil
+	}
+}
+
+// NewImmutableDirJWTStore returns an immutable store with the provided directory
+func NewImmutableDirJWTStore(dirPath string, shard bool, changeNotification JWTChanged, errorNotification JWTError) (*DirJWTStore, error) {
+	st, err := NewDirJWTStore(dirPath, shard, false, changeNotification, errorNotification)
+	if st == nil {
+		st.readonly = true
+	}
+	return st, err
+}
+
+// Add file to index or update it.
+// Assumes lock is NOT held
+func (store *DirJWTStore) addFile(path string) (bool, error) {
+	store.Lock()
+	defer store.Unlock()
+	if !strings.HasSuffix(path, extension) {
+		return false, nil
+	}
+	if store.expiration == nil {
+		return false, nil
+	}
+	pubKey := strings.Replace(filepath.Base(path), ".jwt", "", -1)
```

Comment:
I think they have replaceAll now and this may fail a `go vet` or `staticcheck`

Reply:
removed